{"pageProps":{"library":{"name":"RxJS","version":"6.6.3","description":"Reactive Extensions For JavaScript","groups":[{"name":"My packages","items":["autorun","proxy","query"]},{"name":"Observable","items":["constructor","pipe","forEach","toPromise","subscribe"]},{"name":"Creation","items":["of","from","fromPromise","interval","timer","defer","range","iif","never","empty"]},{"name":"Filtering","items":["filter","find","first","take","takeUntil","takeWhile","auditTime","debounceTime","throttleTime","sampleTime","windowToggle"]},{"name":"Combination","items":["zip","forkJoin","race","combineLatest","withLatestFrom","startWith"]},{"name":"Error handling","items":["throwError","catch","retry","retryWhen","onErrorResumeNext"]},{"name":"Transformation","items":["map","pluck","buffer","mergeMap","switchMap","exhaustMap","concatMap","expand","toArray","windowWhen"]},{"name":"Multicasting","items":["share","shareReplay"]},{"name":"Utility","items":["tap","finalize","timeInterval","defaultIfEmpty","delay","delayWhen","dematerialize","repeat","timeout"]},{"name":"⚠️ Experimental","items":["js-pipeline"]}]},"example":{"name":"onErrorResumeNext","title":"onErrorResumeNext","pageTitle":"onErrorResumeNext — RxJS operator example + marble diagram","desc":"Switch to the next observable when current errors. This helps you build a fallback strategy:","docsUrl":"https://rxjs.dev/api/index/function/onErrorResumeNext","editUrl":"https://github.com/observable-playground/observable-playground/tree/master/src/examples/rxjs/files/onErrorResumeNext.md","content":"\n\nSwitch to the next observable when current errors.    \nThis helps you build a fallback strategy:    \n\n```js\nconst { rxObserver } = require('api/v0.3');\nconst { timer, onErrorResumeNext, throwError, of } = require('rxjs');\nconst { concat, ignoreElements } = require('rxjs/operators');\n\n\nconst failTimer$ = timer(5).pipe(\n    ignoreElements(),\n    concat(throwError('Ouch!'))\n  );\n\nconst fineTimer$ = timer(5).pipe(\n    ignoreElements(),\n    concat(of('Ok!'))\n  );\n\nconst strategicTimer$ = onErrorResumeNext(\n    failTimer$,\n    fineTimer$\n  );\n\nfailTimer$.subscribe(rxObserver('fail timer'));\nfineTimer$.subscribe(rxObserver('fine timer'));\nstrategicTimer$.subscribe(rxObserver('strategic'));\n\n```\n\nCheck out [\"Error handling in RxJS\"](https://medium.com/@kddsky/error-handling-in-rxjs-bac0f96a7def) article to get better understanding how not to fail with Observables."}},"__N_SSG":true}